src/seafile-applet.cpp

```diff
@@ -279,6 +279,20 @@ void SeafileApplet::errorAndExit(const QString& error)
     QCoreApplication::exit(1);
 }
 
+void SeafileApplet::restartApp()
+{
+    if (in_exit_ || QCoreApplication::closingDown()) {
+        return;
+    }
+
+    in_exit_ = true;
+
+    QStringList args = QApplication::arguments();
+    args.removeFirst();
+    QProcess::startDetached(QApplication::applicationFilePath(), args);
+    QCoreApplication::quit();
+}
+
```

Comment:
Currently when seafile-applet starts up, it would check if these is already another "seafile-applet" process running, and if so, tells the user "seafile is already running", and then exits. Is it a possible problem that the old process has not exited yet when the newly started process starts up?